test/Common/MoistThermodynamics/profiles.jl

```diff
@@ -0,0 +1,124 @@
+#### Tested moist thermodynamic profiles
+
+#=
+This file contains functions to compute all of the
+thermodynamic _states_ that MoistThermodynamics is
+tested with in runtests.jl
+=#
+
+
+"""
+    fixed_lapse_rate(
+        param_set::AbstractParameterSet,
+        z::FT,
+        T_surface::FT,
+        T_min::FT,
+        ) where {FT <: AbstractFloat}
+
+Fixed lapse rate hydrostatic reference state
+"""
+function fixed_lapse_rate(
+    param_set::AbstractParameterSet,
+    z::FT,
+    T_surface::FT,
+    T_min::FT,
+) where {FT <: AbstractFloat}
+    _grav::FT = grav(param_set)
+    _cp_d::FT = cp_d(param_set)
+    _R_d::FT = R_d(param_set)
+    _MSLP::FT = MSLP(param_set)
+    Γ = _grav / _cp_d
+    z_tropopause = (T_surface - T_min) / Γ
+    H_min = _R_d * T_min / _grav
+    T = max(T_surface - Γ * z, T_min)
+    p = _MSLP * (T / T_surface)^(_grav / (_R_d * Γ))
+    T == T_min && (p = p * exp(-(z - z_tropopause) / H_min))
+    ρ = p / (_R_d * T)
+    return T, p, ρ
+end
+
+"""
+    tested_profiles(param_set, n::Int, ::Type{FT})
+
+A range of input arguments to thermodynamic state constructors
+
+ - `param_set` an `AbstractParameterSet`, see the [`MoistThermodynamics`](@ref) for more details
+ - `z_all` altitude
+ - `e_int` internal energy
+ - `ρ` (moist-)air density
+ - `q_tot` total specific humidity
+ - `q_pt` phase partition
+ - `T` air temperature
+ - `θ_liq_ice` liquid-ice potential temperature
+
+that are tested for convergence in saturation adjustment.
+
+Note that the output vectors are of size ``n*n_RS``, and they
+should span the input arguments to all of the constructors.
+"""
+function tested_profiles(
+    param_set::AbstractParameterSet,
+    n::Int,
+    ::Type{FT},
+) where {FT}
+
+    n_RS1 = 10
+    n_RS2 = 20
+    n_RS = n_RS1 + n_RS2
+    z_range = range(FT(0), stop = FT(2.5e4), length = n)
+    relative_sat1 = range(FT(0), stop = FT(1), length = n_RS1)
+    relative_sat2 = range(FT(1), stop = FT(1.02), length = n_RS2)
+    relative_sat = [relative_sat1..., relative_sat2...]
+    T_min = FT(150)
+    T_surface = FT(350)
+
+    T = zeros(FT, n, n_RS)
+    p = zeros(FT, n, n_RS)
+    ρ = zeros(FT, n, n_RS)
+    RS = zeros(FT, n, n_RS)
+    z_all = zeros(FT, n, n_RS)
+    for i in eachindex(z_range)
+        for j in eachindex(relative_sat)
+            args = fixed_lapse_rate(param_set, z_range[i], T_surface, T_min)
+            k = CartesianIndex(i, j)
+            z_all[k] = z_range[i]
+            T[k] = args[1]
```

Comment:
```suggestion
            T_virt[k] = args[1]
```
And then you will need to add a nonlinear solve here to get temperature from virtual temperature, by solving
`R_m(q)*T − R_d*T_virt = 0` (Eq. 7.7 of atmosphere design doc). This will require adding an additional root solve in MoistThermodynamics.

Reply:
This is what you'd have to do to get a test set that is thermodynamically consistent and hydrostatic. But there's a much easier way to test that drops the hydrostatic requirements (which doesn't matter here anyway). I'll outline it below.